### PR TITLE
fix(plugin-people): modify people.list to finish batching when notFou…

### PR DESCRIPTION
**Pull request by good-joe**
…ndIds exists

The total batched PII requests are equal to the number of returned PIIs (body.items) and the list of missing PIIs (body.notFoundIds). people.list must resolve both items and notFoundIds to complete the batch and ultimately resolve the requested PIIs.
Previously notFoundIds was ignored and the batch always failed with a timeout.

# Pull Request Template

## Description

On HTTP success the people-batcher now calls handleItemSuccess for each element of notFoundIds, allowing batch to fully resolve all requested ids.
People.list returns an empty list when http errors are encounted (bad request when *no* items are returned, even if notFoundIds could have been populted).
Fixes # (issue)
#846 
## Type of Change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)

## Test Coverage

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Requested a mix of known and unknown ids to test notFoundIds were handled
- [ ] Requested only unknown ids to test no items returned, http request failure

**Test Configuration**:
* SDK Version 
* Node/Browser Version 8.6
* NPM Version 5.7.1

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: packages/node_modules/@ciscospark/plugin-people/src/people.js [hunk 1]
```diff
@@ -147,7 +153,19 @@ const People = SparkPlugin.extend({
...
+        // batcher appends the not founds results contiguously after all of the founds.
+        .then((results) => {
+          let idx = results.length - 1;
+          while (idx >= 0 && !results[idx]) {
+            idx -= 1;
+          }
+          return (idx >= 0 && results.slice(results.length - idx)) || [];
+        })
+        // return an empty list on errors
+        .catch(() => []);
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/plugin-people/src/people.js]: Should we use the logger to log the errors that we are catching?

File: packages/node_modules/@ciscospark/plugin-people/test/integration/spec/people.js [hunk 2]
```diff
@@ -73,6 +77,33 @@ describe('plugin-people', function () {
...
+        .then((peopleResponse) => {
+          assert.equal(peopleResponse.length, 3);
+          assert.isPerson(peopleResponse[0]);
+          assert.isPerson(peopleResponse[1]);
+          assert.isPerson(peopleResponse[2]);
+          assert.equal(peopleResponse[0].emails[0], mccoy.email);
+          assert.equal(peopleResponse[1].emails[0], spock.email);
+          assert.equal(peopleResponse[2].emails[0], bones.email);
+        }));
+      // list([VALID_ID, INVALID_ID]) -> valid response with VALID_ID user, and INVALID_ID in notFoundIds
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/plugin-people/test/integration/spec/people.js]: (nitpick) can we get an empty space between `it` blocks?

File: packages/node_modules/@ciscospark/plugin-people/src/people-batcher.js [hunk 3]
```diff
@@ -18,8 +18,11 @@ const PersonUUIDRequestBatcher = Batcher.extend({
    * @returns {Promise}
    */
   handleHttpSuccess(res) {
+    // combine item and notFoundIds lists
+    const notFounds = res.body.notFoundIds || [];
     return Promise.all(res.body.items.map((personResponse) =>
-      this.handleItemSuccess(personResponse.id, personResponse)));
+      this.handleItemSuccess(personResponse.id, personResponse))
+      .concat(notFounds.map((id) => this.handleItemSuccess(id))));
```
Comment (adamweeks) [inline packages/node_modules/@ciscospark/plugin-people/src/people-batcher.js]: If we are going through the work to remove these in `people.js`, what benefit do we get from `.concat`ing them?

File: packages/node_modules/@ciscospark/plugin-people/test/integration/spec/people.js (hunk 2 above)
Comment (good-joe) [reply packages/node_modules/@ciscospark/plugin-people/test/integration/spec/people.js]: +1

File: packages/node_modules/@ciscospark/plugin-people/src/people.js (hunk 1 above)
Comment (good-joe) [reply packages/node_modules/@ciscospark/plugin-people/src/people.js]: will do

File: packages/node_modules/@ciscospark/plugin-people/src/people-batcher.js (hunk 3 above)
Comment (good-joe) [reply packages/node_modules/@ciscospark/plugin-people/src/people-batcher.js]: Way to get to the heart of it Adam. In truth I wasn't sure what batch needed underneath to know when a request had finished. Now after looking through underlying code, what matters is all the request made in people.js line 50 resolve. The return value from handleHTTPSuccess doesn't matter at all for our purposes. 

So a misunderstanding, good catch Adam. I'll move the notFoundIds handling to a resolver after PeopleBatcher line 23 Promise.all

**Pull request closed without merging**